### ISSUE TITLE
⌨ Skill browser submit button loses focus when submitting

### ISSUE THREAD

**Issue by esizer** [GCTC-NTGC/gc-digital-talent#13254]
## ⌨ Accessibility Issue

The submit button in the skill browser dialog loses focus when you interact with it. This is a problem if you submit with an error, you lose focus on the current item and have to cycle back around to fix the error.

I believe this is because we are disabling the submit button on a submission attempt.

### WCAG 2.1 Criteria

Technically you are allowed to disable buttons, it is just [highly discouraged](https://adrianroselli.com/2024/02/dont-disable-form-controls.html#Submit).

## 🦋 Expected Behaviour

Your focus remains intact while submitting the form.

## 🕵️ Details

- **Browser**: Google chrome, safari, firefox
- **Assistive Technology**: None, just the keyboard

## 📋 Steps to Reproduce

1. Login as an applicant `applicant@test.com`
2. Navigate to create an experience
3. Open the "Add skill" dialog
4. Using only the keyboard, attempt to submit the form with an error
5. Observe the focus is lost on the submit button and you need to tab to get back

## 🙋‍♀️ Proposed Solution

Rather than disabling an input, prevent a mutation or callback from happening while one is in flight. 

## ✅ Acceptance Criteria

- [ ] Focus remains on submit button while form is submitting
- [ ] Duplicate submissions cannot occur
